src/operator/subgraph/dnnl/dnnl_transformer.cc

```diff
@@ -29,54 +29,78 @@
 #include "operator/subgraph/common.h"
 #include "dnnl_transformer-inl.h"
 
-// 3 tensors within one (queries key values) =
+// 3 tensors within one (queries key values)
 #define QKV_NUM 3
 
 namespace mxnet {
 namespace op {
 
 DMLC_REGISTER_PARAMETER(DNNLSelfAttParam);
 
+template <bool with_split>
 static bool SgDNNLSelfAttShape(const NodeAttrs& attrs,
                                mxnet::ShapeVector* in_shape,
                                mxnet::ShapeVector* out_shape) {
-  const auto& params = nnvm::get<DNNLSelfAttParam>(attrs.parsed);
-  auto qkv_shape     = in_shape->at(0);
-  CHECK_EQ(qkv_shape.ndim(), 3U)
+  const auto& params        = nnvm::get<DNNLSelfAttParam>(attrs.parsed);
+  unsigned int in_shape_num = 1;
+  auto in_shape_0           = in_shape->at(0);
+  auto in_shape_1           = in_shape_0;  // in with_split mode there is only one input
```

Comment:
Maybe simply with split/without split, without node and underscore?
```suggestion
  auto in_shape_1           = in_shape_0;  // with split there is only one input
```